src/main/java/org/plumelib/util/CombinationIterator.java

```diff
@@ -52,27 +56,37 @@ public class CombinationIterator<T> implements Iterator<List<T>> {
    */
   @SuppressWarnings({"rawtypes", "unchecked"}) // for generic array creation
   public CombinationIterator(Collection<? extends Collection<T>> collectionsOfCandidates) {
-    int size = collectionsOfCandidates.size();
-    // Just like collectionsOfCandidates, but indexable.
-    ArrayList<? extends Collection<T>> listOfCollectionsOfCanditates =
-        new ArrayList<>(collectionsOfCandidates);
-    listsOfCandidates = new ArrayList[size];
-    iterators = new Iterator[size];
-    combinationSize = size;
-    nextValue = (combinationSize == 0 ? null : new ArrayList<>(collectionsOfCandidates.size()));
-
-    for (int i = 0; i < combinationSize; i++) {
-      Collection<T> userSuppliedCandidates = listOfCollectionsOfCanditates.get(i);
+    this.listOfCollectionsOfCanditates = new ArrayList<>(collectionsOfCandidates);
+
+    final int n = this.listOfCollectionsOfCanditates.size();
+
+    // Create arrays, then assign to fields. Suppress ONLY the cast-unsafe warnings here.
+    @SuppressWarnings("samelen:cast.unsafe")
+    List<T> @SameLen("this.listOfCollectionsOfCanditates") [] listsTmp =
+        (List<T> @SameLen("this.listOfCollectionsOfCanditates") []) new List<?>[n];
+    this.listsOfCandidates = listsTmp;
+
+    @SuppressWarnings("samelen:cast.unsafe")
+    Iterator<T> @SameLen("this.listOfCollectionsOfCanditates") [] itersTmp =
+        (Iterator<T> @SameLen("this.listOfCollectionsOfCanditates") []) new Iterator<?>[n];
+    this.iterators = itersTmp;
+
+    this.combinationSize = n;
+    this.nextValue = (n == 0 ? null : new ArrayList<>(n));
+
+    for (int i = 0; i < n; i++) {
+      Collection<T> userSuppliedCandidates = this.listOfCollectionsOfCanditates.get(i);
 
       List<T> candidates = new ArrayList<>(userSuppliedCandidates);
-      listsOfCandidates[i] = candidates;
+      this.listsOfCandidates[i] = candidates;
       Iterator<T> it = candidates.iterator();
-      iterators[i] = it;
-      if (nextValue != null) {
+      this.iterators[i] = it;
+
+      if (this.nextValue != null) {
         if (it.hasNext()) {
-          nextValue.add(it.next());
+          this.nextValue.add(it.next());
```

Comment:
Please do not make unrelated changes; they obscure the logic of the pull request.  If you want to make this change, do it in a separate pull request.  It should not appear in this pull request.